tests/kohana/ArrTest.php

```diff
@@ -443,6 +445,10 @@ public function provider_path()
 			// Path as array, issue #3260
 			array($array['users'][2]['name'], $array, array('users', 2, 'name')),
 			array($array['object']['iterator'], $array, 'object.iterator'),
+            // ArrayObject
+            array($array['kohana'], $array_object, 'kohana'),
+            array($array['foobar']['definition'], $array_object, 'foobar.definition'),
+            array($array['users'][1]['name'], $array_object, 'users.1.name'),
```

Comment:
Prob also worth adding an explicit test for the undefined/default case too. Shouldn't differ from `array` but again just in case.

Reply:
See 2680cd1 I think that covers the default case explicitly NULL and added the default array case too just to be sure